### PR TITLE
configurable service account and existingSecret for externalPostgresql

### PR DESCRIPTION
**Pull request by kimxogus**
For EKS IRSA, this is necessary

### CODE REVIEW

**Comment by kmmanto** [comment]
Would like to see this merged.

File: sentry/templates/deployment-relay.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ spec:
 {{ toYaml .Values.relay.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-relay.yaml:45]: make it optional

File: sentry/templates/deployment-sentry-cron.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ spec:
 {{ toYaml .Values.sentry.cron.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-sentry-cron.yaml:35]: make it optional

File: sentry/templates/deployment-sentry-ingest-consumer.yaml [hunk 3]
```diff
@@ -43,6 +43,7 @@ spec:
 {{ toYaml .Values.sentry.ingestConsumer.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-sentry-ingest-consumer.yaml:46]: make it optional

File: sentry/templates/deployment-sentry-post-process-forwarder.yaml [hunk 4]
```diff
@@ -38,6 +38,7 @@ spec:
 {{ toYaml .Values.sentry.postProcessForward.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-sentry-post-process-forwarder.yaml:44]: make it optional

File: sentry/templates/deployment-sentry-web.yaml [hunk 5]
```diff
@@ -36,6 +36,7 @@ spec:
 {{ toYaml .Values.sentry.web.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-sentry-web.yaml:39]: make it optional

File: sentry/templates/deployment-sentry-worker.yaml [hunk 6]
```diff
@@ -34,6 +34,7 @@ spec:
 {{ toYaml .Values.sentry.worker.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-sentry-worker.yaml:37]: make it optional

File: sentry/templates/deployment-snuba-api.yaml [hunk 7]
```diff
@@ -33,6 +33,7 @@ spec:
 {{ toYaml .Values.snuba.api.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-snuba-api.yaml:36]: make it optional

File: sentry/templates/deployment-snuba-consumer.yaml [hunk 8]
```diff
@@ -40,6 +40,7 @@ spec:
 {{ toYaml .Values.snuba.consumer.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-snuba-consumer.yaml:43]: make it optional

File: sentry/templates/deployment-snuba-outcomes-consumer.yaml [hunk 9]
```diff
@@ -40,6 +40,7 @@ spec:
 {{ toYaml .Values.snuba.outcomesConsumer.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-snuba-outcomes-consumer.yaml:43]: make it optional

File: sentry/templates/deployment-snuba-replacer.yaml [hunk 10]
```diff
@@ -40,6 +40,7 @@ spec:
 {{ toYaml .Values.snuba.replacer.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-snuba-replacer.yaml:43]: make it optional

File: sentry/templates/deployment-snuba-sessions-consumer.yaml [hunk 11]
```diff
@@ -40,6 +40,7 @@ spec:
 {{ toYaml .Values.snuba.sessionsConsumer.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-snuba-sessions-consumer.yaml:43]: make it optional

File: sentry/templates/deployment-snuba-transactions-consumer.yaml [hunk 12]
```diff
@@ -40,6 +40,7 @@ spec:
 {{ toYaml .Values.snuba.transactionsConsumer.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-snuba-transactions-consumer.yaml:43]: make it optional

File: sentry/templates/deployment-symbolicator.yaml [hunk 13]
```diff
@@ -33,6 +33,7 @@ spec:
 {{ toYaml .Values.symbolicator.api.podLabels | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ include "sentry.serviceAccountName" . }}
```
Comment (Mokto) [inline sentry/templates/deployment-symbolicator.yaml:36]: make it optional

File: sentry/values.yaml [hunk 14]
```diff
@@ -50,8 +50,8 @@ relay:
     minReplicas: 2
```
Comment (Mokto) [inline sentry/values.yaml:50]: Please remove all your changes done by your linter.

File: sentry/values.yaml [hunk 15]
```diff
@@ -470,6 +471,10 @@ config:
   relay: |
     # No YAML relay config given
 
+serviceAccount:
+  create: true
```
Comment (Mokto) [inline sentry/values.yaml:522]: Should be false per default.

**Review by Mokto — CHANGES_REQUESTED** [review]

File: sentry/values.yaml (hunk 15 above)
Comment (Mokto) [inline sentry/values.yaml:522]: Also I believe you're not really creating the proper service account?

**Review by Mokto — CHANGES_REQUESTED** [review]

**Comment by z0rc** [comment]
@kimxogus please check linked PR, support of serviceAccounts was added recently. If you're still interested in existingSecret for externalPostgresql, please update the PR or factor those changes into new PR.

**Comment by Mokto** [comment]
I'm closing this PR as it's really old. Feel free to reopen it if needed.

**Pull request closed without merging**